addons/crm/models/crm_lead.py

```diff
@@ -595,8 +595,18 @@ def write(self, vals):
         if any(field in ['active', 'stage_id'] for field in vals):
             self._handle_won_lost(vals)
 
+        from_lead_values = dict([lead.id, [lead.stage_id.id, lead.active]] for lead in self)
         write_result = super(Lead, self).write(vals)
 
+        # Recompute the probabilities for a lead if its stage or activation has changed
+        # This allows to take into account the table update in _handle_won_lost
+        self.flush(['probability', 'automated_probability'])
```

Comment:
Hello @dbeguin , So this seems to pass all the tests, but the flush does disturb the query count in some perf tests. Any thoughts? Cheers

Reply:
this seems ok to upgrade the query counter in this case, as we need the lush to make this work properly.